app/connectors/AuthConnector.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.WSHttp
+import model.{BearerToken, LoginDetails, Role, SuccessfulAuthentication}
+import play.api.libs.json.Json
+import uk.gov.hmrc.play.config.ServicesConfig
+import uk.gov.hmrc.play.http._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+object AuthConnector extends AuthConnector with ServicesConfig {
+  override val internalUserAuthBaseUrl: String = s"${baseUrl("auth")}/auth/authenticate/user"
+  override val http = WSHttp
+}
+
+trait AuthConnector {
+
+  private case class AuthBackendResponse(access_token: BearerToken, group: Option[String], roles: Option[Set[Role]]) {
+    def toAuthSuccess(loginDetails: LoginDetails) = SuccessfulAuthentication(access_token, loginDetails.userName, roles)
+  }
+
+  private object AuthBackendResponse {
+    implicit val reads = Json.reads[AuthBackendResponse]
+  }
+
+  class InvalidCredentials extends RuntimeException("Login failed")
+
+  val internalUserAuthBaseUrl: String
+  val http: HttpPost with HttpGet with HttpDelete
+
+  def login(loginDetails: LoginDetails)(implicit hc: HeaderCarrier): Future[SuccessfulAuthentication] =
+    http.POST[LoginDetails, AuthBackendResponse](internalUserAuthBaseUrl, loginDetails)
+      .map(_.toAuthSuccess(loginDetails))
+      .recover {
+        case e: Upstream4xxResponse if e.upstreamResponseCode == 401 => throw new InvalidCredentials
+      }
```

Comment:
We could use `.recoverWith` and a `Future.failed(new InvalidCredentials)` here.  In this case, it is obvious that the future will catch the error and convert it to a failed future - but I think being explicit about it is a good habit to get into. 

That way, we'll avoid some of the issues we've had with accidentally getting a thrown error instead of a failed future we've had in the past..